### DISCUSSION TITLE
Postgresql service is stopped. But why there is no error when trying to connect?

### DISCUSSION THREAD

**Discussion by ankurdeka** [jackc/pgx#1324] [Q&A] [answered]
```go
var db *pgxpool.Pool
var ctx, cancel = context.WithTimeout(context.Background(), 15*time.Second)

func init() {
	err := godotenv.Load(".env")
	if err != nil {
		fmt.Println("Could not load .env file")
		os.Exit(1)
	}
	db, err = pgxpool.New(ctx, os.Getenv("DATABASE_URL"))
	if err != nil {
		fmt.Println("Could not connect to database:", err)
		cancel()
		os.Exit(1)
	}
}
```

**Comment by ankurdeka** [comment, marked as answer]
Solved.

There are 2 options. One is `Ping` and the other is `Acquire`.
Which one is the best option to check if the connection is successful?

**Reply by jackc** [reply]
In pgx v5 creating a pool does not automatically connect. The reason for this is a service shouldn't die / fail to start if the database it is trying to connect to is temporarily unavailable.

Since you only want to test the connection `Ping` would be the appropriate method.
